doc/services/llext/config.rst

```diff
@@ -0,0 +1,102 @@
+Configuration
+#############
+
+The following Kconfig options are available for the LLEXT subsystem:
+
+.. _llext_kconf_heap:
+
+Heap size
+----------
+
+The LLEXT subsystem requires a heap to be allocated for storing extension
+related data. The size of this heap can be configured using the Kconfig option
+:kconfig:option:`CONFIG_LLEXT_HEAP_SIZE`.
+
+.. note::
+
+   When :ref:`user mode <usermode_api>` is enabled, the heap size must be
+   large enough to allow the extension sections to be allocated with the
+   alignment required by the architecture.
+
+.. _llext_kconf_type:
+
+ELF object type
+---------------
+
+The LLEXT subsystem supports loading different types of extensions; the type
+can be set by choosing among the following Kconfig options:
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_OBJECT`
+
+        Build and expect relocatable files as binary object type for the LLEXT
+        subsystem. A single compiler invocation is used to generate the object
+        file.
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_RELOCATABLE`
+
+        Build and expect relocatable (partially linked) files as the binary
+        object type for the LLEXT subsystem. These object files are generated
+        by the linker by combining multiple object files into a single one.
+
+:kconfig:option:`CONFIG_LLEXT_TYPE_ELF_SHAREDLIB`
+
+        Build and expect shared libraries as binary object type for the LLEXT
+        subsystem. The standard linking process is used to generate the shared
+        library from multiple object files.
+
+        This is not currently supported on ARM architectures.
+
+.. _llext_kconf_storage:
+
+Minimize allocations
+--------------------
+
+The LLEXT subsystem loading mechanism, by default, uses a seek/read abstraction
+and copies all data into allocated memory; this is done to allow the extension
+to be loaded from any storage medium. Sometimes, however, data is already in a
+buffer in RAM and copying it is not necessary. In this case, the option
+:kconfig:option:`CONFIG_LLEXT_STORAGE_WRITABLE` can be enabled to allow the
```

Comment:
ditto re: making this a top-level "term" to stand out more / look more like the ELF Object type section